enterprise/metrics/src/main/java/org/neo4j/metrics/source/ClusterMetrics.java

```diff
@@ -75,6 +96,30 @@ public Long getValue()
                     return monitor.lastAppliedTxId;
                 }
             } );
+
+        }
+    }
+
+    private boolean canResolveClusterMembersDependency()
+    {
+        try
+        {
+            clusterMembers = dependencyResolver.resolveDependency( ClusterMembers.class );
+        }
+        catch(IllegalArgumentException e)
+        {
+            logService.getUserLog( getClass() ).warn("Cluster metrics was enabled but the appropriate " +
+                    "cluster dependencies were not available. Disabling cluster member metrics.");
+        }
+        return clusterMembers != null;
+    }
+
+    private void registerClusterMemberMetrics()
+    {
+        if(clusterMembers != null)
```

Comment:
we already checking availability of clusterMembers before invoking this method and will not invoke registration if we failed at all